### DISCUSSION TITLE
extension utilized cli command (using the terminal)

### DISCUSSION THREAD

**Discussion by guyarieli** [microsoft/vscode-discussions#829] [Extension Development QnA] [answered]
My extension is utilizing a cli command using the terminal. The extension code is not dependent on the cli tool.

What is the right way to include such a package?

I have tried the following:

const path = extensionPath + "/node_modules/some_module/bin/client.js";
terminalContainer.terminal.sendText("node " + path + " " + workspaceRoot");
...

But I realized the node_modules will not be available after packaging my extension.
Thanks,
Guy

**Comment by r3m0t** [comment, marked as answer]
Removal of `node_modules` is due to `.vscodeignore`. You can change it. Or use a bundler like Webpack or esbuild to create a .js that includes its dependencies. In that case "entry" in Webpack config would be set to bin/client.js.

https://webpack.js.org/configuration/entry-context/
